ibis/backends/tests/test_window.py

```diff
@@ -1286,3 +1286,78 @@ def test_duplicate_ordered_sum(con):
     # final position, since the *output* order doesn't depend on ORDER BY
     # provided by the user
     assert result[2:] in ([60, 100], [70, 100], [100, 60], [100, 70])
+
+
+mark_sqlite_no_collect = pytest.mark.never(
+    ["sqlite"],
+    raises=com.OperationNotDefinedError,
+    reason="sqlite doesn't have array types",
+)
+mark_sqlite_no_ignore_nulls = pytest.mark.never(
+    "sqlite",
+    reason="sqlite doesn't support the `IGNORE NULLS` option for first_value()",
+)
+
+
+@pytest.mark.notimpl(
+    ["polars"],
+    raises=com.OperationNotDefinedError,
+    reason="window functions aren't yet implemented for the polars backend",
+)
+@pytest.mark.parametrize(
+    "aggname,include_null,expected",
+    [
+        ("first", True, [None, None, 6, 6]),
+        pytest.param(
+            "first",
+            False,
+            [5, 5, 6, 6],
+            marks=[mark_sqlite_no_ignore_nulls],
+        ),
+        ("last", True, [5, 5, None, None]),
+        pytest.param(
+            "last",
+            False,
+            [5, 5, 6, 6],
+            marks=[mark_sqlite_no_ignore_nulls],
+        ),
+        pytest.param(
+            "collect",
+            True,
+            [[None, 5], [None, 5], [6, None], [6, None]],
+            marks=[mark_sqlite_no_collect],
+        ),
+        pytest.param(
+            "collect",
+            False,
+            [[5], [5], [6], [6]],
+            marks=[
+                # pytest.mark.notyet(
+                #     ["datafusion"],
+                #     raises=AssertionError,
+                #     reason="gives the same as for include_null=True, but should be different",
+                # ),
```

Comment:
is this comment needed?

Reply:
no I should remove it. I still need to sort out what all the cloud backends say (I can't seem to get a local dev env set up correctly with cloud backends so I have to use the CI 😢  ) and update the marks